core/impl/src/main/java/com/blazebit/persistence/impl/function/datediff/quarter/MSSQLQuarterDiffFunction.java

```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright 2014 - 2019 Blazebit.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.blazebit.persistence.impl.function.datediff.quarter;
+
+/**
+ * @author Jan-Willem Gmelig Meyling
+ * @since 1.4.0
+ */
+public class MSSQLQuarterDiffFunction extends QuarterDiffFunction {
+
+    public MSSQLQuarterDiffFunction() {
+        super("(select sign(i1) * floor(abs(i1)) from (values (datediff(mm, ?1, ?2)/3)) as temp(i1))");
```

Comment:
Shouldn't a cast to int be enough?

Reply:
Db2, psql and oracle round to the closest int when casting, which is not appropiate here